paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentOptionsScreen.kt

```diff
@@ -0,0 +1,114 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidViewBinding
+import com.stripe.android.link.ui.verification.LinkVerificationDialog
+import com.stripe.android.paymentsheet.PaymentOptionsViewModel
+import com.stripe.android.paymentsheet.databinding.FragmentPaymentOptionsPrimaryButtonBinding
+import com.stripe.android.ui.core.elements.H4Text
+import com.stripe.android.uicore.StripeTheme
+import com.stripe.android.uicore.stripeColors
+import com.stripe.android.uicore.text.Html
+
+@Composable
+internal fun PaymentOptionsScreen(
+    viewModel: PaymentOptionsViewModel,
+    modifier: Modifier = Modifier,
+) {
+    val scrollState = rememberScrollState()
+
+    val didScroll = remember(scrollState.value) {
+        scrollState.value > 0
+    }
+
+    val elevation by animateDpAsState(targetValue = if (didScroll) 8.dp else 0.dp)
+
+    Column {
+        Surface(elevation = elevation) {
+            PaymentSheetTopBar(viewModel)
+        }
```

Comment:
We can’t use a `Scaffold` for this, as it seems to force a full-screen layout when we want the bottom sheet to have adaptive height.